assets/js/script.js

```diff
@@ -0,0 +1,333 @@
+      timer -= 5;
+      // render error alert with message and status
+      renderAnswerAlarm("Oops, be careful! You have less time now...");
+    }
+
+    // if statement to check the last question or not
+    if (questionIndex < questions.length - 1) {
+      questionIndex += 1;
+      removeQuestionSection();
+      renderQuestionSection();
+    } else {
+      // set quizComplete to true and then render form and stop the timer
+      removeQuestionSection();
+      renderGameOver();
+    }
+    const clearAlarm = () => {
+      if (document.getElementById("answerAlarm")) {
+        document.getElementById("answerAlarm").remove();
+      }
+      clearTimeout(alarmTimer);
+    };
+
+    const alarmTimer = setTimeout(clearAlarm, 1500);
+  }
+};
+
+const removeStartSection = () => {
+  mainSection.removeChild(startContainer);
+};
+
+const removeQuestionSection = () => {
+  const questionSection = document.getElementById("question-section");
+  mainSection.removeChild(questionSection);
+};
+const renderQuestionSection = () => {
+  // get each question by get the questions array
+  const question = questions[questionIndex];
+
+  // creat section element
+  const questionSection = document.createElement("section");
+  questionSection.setAttribute("class", "question-section");
+  questionSection.setAttribute("id", "question-section");
+
+  // create h1 title
+  const h1 = document.createElement("h1");
+  h1.setAttribute("class", "question-section-title");
+  h1.textContent = `Question ${questionIndex + 1}:`;
+
+  // creat h2 info
+  const h2 = document.createElement("h2");
+  h2.setAttribute("class", "question-section-info");
+  h2.textContent = question.question;
+
+  // creat ul info
+  const ul = document.createElement("ul");
+  ul.setAttribute("class", "answers");
+
+  // creat answers using forEach function
+  const createAppendAnswer = (answer) => {
+    const li = document.createElement("li");
+    const btn = document.createElement("button");
+    btn.setAttribute("class", "answer-button");
+    btn.setAttribute("data-value", answer);
+    btn.textContent = answer;
+    li.append(btn);
+    ul.append(li);
+  };
+
+  question.answers.forEach(createAppendAnswer);
+
+  // append section and main
+  questionSection.append(h1, h2, ul);
+  mainSection.append(questionSection);
+
+  // add click event listener on question-section to check answer
+  questionSection.addEventListener("click", validateAnswer);
+};
+
+const renderTimerSection = () => {
+  const timerSpanSection = document.createElement("section");
+  timerSpanSection.setAttribute("id", "timerSpanSection");
+  timerSpanSection.textContent = "Timer: ";
+  const timerSpan = document.createElement("span");
+  timerSpan.setAttribute("id", "timerSpan");
+  timerSpan.textContent = timer;
+  timerSpanSection.append(timerSpan);
+  mainSection.append(timerSpanSection);
+
+  const startTimer = () => {
+    timer -= 1;
+    const timerSpan = document.getElementById("timerSpan");
```

Comment:
could you consider moving this line outside the time maybe put it on line 280...what happens there is that every second the variable is been defined again and again in the memory instead of once

Reply:
updated, thanks